### PR TITLE
Correctly read command pipes in esti/gc runCommand

### LINKED ISSUE #5654: [esti/gc] runCommand can drop command outputs
runCommand in gc_utils_test uses StdoutPipe and StderrPipe.  Godoc for both of these functions [says](https://pkg.go.dev/os/exec#Cmd.StdoutPipe)

> Wait will close the pipe after seeing the command exit, so most callers need not close the pipe themselves. It is thus incorrect to call Wait before all reads from the pipe have completed. For the same reason, it is incorrect to call Run when using StdoutPipe. See the example for idiomatic usage. 

Instead, it should wait for handlePipe to finish reading (and handlePipe should _not_ close its pipe).

### PR DESCRIPTION
**Pull request by arielshaqed**
Fixes #5654.  Uses cmd.Std*Pipe as documented; the previous code might drop some last lines from Spark outputs.

It also ends up reporting the _source_ of each log line from Spark, including the name of the test that ran it.  That's important because all these Sparks run in parallel.  So we now see lines like
```
time="2023-04-09T13:00:10Z" level=info msg="23/04/09 13:00:10 INFO MetricsSystemImpl: Stopping s3a-file-system metrics system..." func=github.com/treeverse/lakefs/esti.handlePipe.func1 file="/home/runner/work/lakeFS/lakeFS/esti/gc_utils_test.go:46" source=gc-gc1 std=err
```
which says that Spark wrote this to **standard error** while running gc-gc1.

### CODE REVIEW

File: esti/gc_utils_test.go [hunk 1]
```diff
@@ -30,25 +32,30 @@ func getDockerArgs(workingDirectory string, localJar string) []string {
...
 			}
-		}()
-	}
+			log(strings.TrimSuffix(str, "\n"))
+		}
+		if err == io.EOF {
+			err = nil
+		}
+		ch <- err
+		pipe.Close()
```
Comment (johnnyaug) [inline esti/gc_utils_test.go:52]: IIUC, this is not needed?

**Review by johnnyaug — APPROVED** [review]
Ouch. See one comment

File: esti/gc_utils_test.go (hunk 1 above)
Comment (arielshaqed) [reply esti/gc_utils_test.go:52]: You're right, not needed here.

Removed, thanks!

**Comment by arielshaqed** [comment]
Thanks!

Pulling...

**Pull request merged**